File: src/BOJ13018.cpp
// https://www.acmicpc.net/problem/13018

#include <iostream>

using namespace std;

int gcd(int x, int y) {
    if (y == 0)
        return x;
    return gcd(y, x % y);
}

void sequence(int n, int k) {
    int d = n - k;
    int* a = new int[n];
    for (int i = 1; i <= n; i++)
        a[i - 1] = i;

    if (n == k) {
        cout << "Impossible" << endl;
        delete a;
        return;
    }
    else if (d == 1) {
        for (int i = 0; i < n; i ++) {
            cout << a[i] << ' ';
        }
        cout << endl;
        delete a;
        return;
    }
    else {
        int temp = 0;
        for (int i = 1; i < d; i++) {
            temp = a[i];
            a[i] = a[i - 1];
            a[i - 1] = temp;
        }
        int j = 0;
        while (true) {
            j = 0;
            for (int i = 1; i < d; i++) {
                if (gcd(i, a[i - 1]) != 1) {
                    for (int i = 1; i < d; i++) {
                        temp = a[i];
                        a[i] = a[i - 1];
                        a[i - 1] = temp;
                    }
                }
                else
                    j++;
            }
            if (j == d-1)
                break;
        }
        for (int i = 0; i < n; i++) {
            cout << a[i] << ' ';
        }
        cout << endl;
        delete a;
        return;
    }
}

int main() {
    int n = 0, k = 0;
    cin >> n >> k;
    sequence(n, k);

    return 0;
}

File: Acmicpc/Python/BOJ13018.py
# https://www.acmicpc.net/problem/13018
# BOJ13018.py

from sys import stdin
from sys import stdout

def sequence(n, k):
    d = n - k
    arr = [i for i in range(1, n+1)]
    if n == k:
        stdout.write('Impossible' + '\n')
    elif d == 1:
        for i in arr:
            stdout.write(str(i) + ' ')
        stdout.write('\n')
    else:
        for i in range(1, d):
            arr[i-1], arr[i] = arr[i], arr[i-1]
        for i in arr:
            stdout.write(str(i) + ' ')
        stdout.write('\n')

if __name__=="__main__":
    n, k = map(int, stdin.readline().split())
    sequence(n, k)
